public.js

```diff
@@ -50,25 +51,46 @@ var init = function(app, schema, sequelize, config) {
 
             /// 1. Am I Facebook Crawler?
             //https://developers.facebook.com/docs/sharing/webmasters/crawler
-            if (/facebookexternalhit|Facebot/.test(req.get('User-Agent')) && parseInt(req.params.abver)) {
+            if (/facebookexternalhit|Facebot/.test(req.get('User-Agent')) && parseInt(req.params.abver) >= 0 ) {
 
-              //What does FB do if you send it a 302 (temporary redirect)?
-              // will it try to visit it again and get a different 302 if visiting again, or will it just
-              // cache it?
-              // If it DOES redirect each time, then /0/ could choose by bandit, itself
-
-              var murl = (req.params.domain + decodeURIComponent(pathname || '/'));
+              var murl = (req.params.domain + decodeURIComponent(pathname || '/').replace(/.fb\d+/,''));
               schema.Metadata.findOne({
-                'where': { 'url':murl.replace(/.fb\d+/,''), 'id':parseInt(req.params.abver)}
+                'where': { 'url':murl, 'id':parseInt(req.params.abver)}
               }).then(function(trial) {
-                //maybe this part should go down by 'function failure'?
                 if (!trial) {
                   if (/testshare/.test(pathname)) {
                     res.render('shareheaders', {
                       'extraProperties': domainInfo.extraProperties || [],
                       'title': "Fooooo",
                       'description': 'basdfasdf',
                     });
+                  } else if (req.params.abver == '0') {
+                    // if facebook is sent a sharebandit URL with no treatment id render best treatment metadata and refresh fb cache
+                      schema.Metadata.findOne({
+                        'where': { 'url':murl },
+                        'order': [[ 'success_count', 'DESC' ]],
+                        'limit': 1
+                      }).then(function(bestTrial) {
+                        if (bestTrial) {
+                          var overOneHourAgo = (date) => {
+                            var HOUR = 1000 * 60 * 60;
+                            var hourAgo = Date.now() - HOUR;
+                            return date < hourAgo;
+                          }
+                          if (overOneHourAgo(bestTrial.updatedAt)) {
```

Comment:
remove this condition entirely here and then updateFacebookCache will get called from a trigger in lambda or whatever.